### ISSUE TITLE
Gradle fails to download latest released version (2.7.9.3)

### ISSUE THREAD

**Issue by innovationhub-asia** [tdlight-team/tdlight-java#61]
Gradle fails to download a version with the following exception:

`Caused by: org.gradle.internal.resolve.ModuleVersionResolveException: Could not resolve it.tdlight:tdlight-java-parent:2.7.9.3.
Caused by: org.gradle.api.internal.artifacts.ivyservice.ivyresolve.parser.MetaDataParseException: inconsistent module metadata found. Descriptor: it.tdlight:tdlight-java-parent:1.0.0.0-SNAPSHOT Errors: bad version: expected='2.7.9.3' found='1.0.0.0-SNAPSHOT'
        at org.gradle.api.internal.artifacts.repositories.metadata.AbstractRepositoryMetadataSource.checkModuleIdentifier(AbstractRepositoryMetadataSource.java:126)
        at org.gradle.api.internal.artifacts.repositories.metadata.AbstractRepositoryMetadataSource.checkMetadataConsistency(AbstractRepositoryMetadataSource.java:116)
        at org.gradle.api.internal.artifacts.repositories.metadata.DefaultMavenPomMetadataSource.parseMetaDataFromResource(DefaultMavenPomMetadataSource.java:69)
        at org.gradle.api.internal.artifacts.repositories.metadata.AbstractRepositoryMetadataSource.parseMetaDataFromArtifact(AbstractRepositoryMetadataSource.java:81)
        at org.gradle.api.internal.artifacts.repositories.metadata.AbstractRepositoryMetadataSource.create(AbstractRepositoryMetadataSource.java:68)
        at org.gradle.api.internal.artifacts.repositories.metadata.DefaultMavenPomMetadataSource.create(DefaultMavenPomMetadataSource.java:40)`

**Comment by Eilip999** [comment]
Same issue here.

**Comment by sayamahendra** [comment]
Same issue here

**Comment by tifoha** [comment]
Same issue here

**Comment by cavallium** [comment]
I'm about to fix this bug, in the next release read carefully the dependency installation instructions because they will be different.

**Comment by innovationhub-asia** [comment]
Still does not work, after applying exact suggested configuration.

Another step that was omited is adding plugin:

`id "io.spring.dependency-management" version "1.0.11.RELEASE"`

without with the configuration 

```
dependencyManagement {
    imports {
        mavenBom 'org.springframework.cloud:spring-cloud-dependencies:Edgware.SR4'
    }
}
```

is unknown.

2021-12-08T23:03:56.427+0800 [WARN] [org.gradle.plugins.ide.internal.resolver.UnresolvedIdeDependencyHandler] Could not resolve: it.tdlight:tdlight-java-bom:2.7.10.0
2021-12-08T23:03:56.427+0800 [DEBUG] [org.gradle.plugins.ide.internal.resolver.UnresolvedIdeDependencyHandler] Could not resolve: it.tdlight:tdlight-java-bom:2.7.10.0
org.gradle.internal.resolve.ModuleVersionResolveException: Could not resolve it.tdlight:tdlight-java-bom:2.7.10.0.
Caused by: org.gradle.internal.resolve.ModuleVersionResolveException: Could not resolve it.tdlight:tdlight-java-bom:2.7.10.0.
Caused by: org.gradle.api.internal.artifacts.ivyservice.ivyresolve.parser.MetaDataParseException: inconsistent module metadata found. Descriptor: it.tdlight:tdlight-java-bom:1.0.0.0-SNAPSHOT Errors: bad version: expected='2.7.10.0' found='1.0.0.0-SNAPSHOT'

**Comment by cavallium** [comment]
Fixed in 2.7.10.3

**Comment by cavallium** [comment]
You should remove the `dependencyManagement` part.

It was a leftover that I forgot to remove from the guide.

**Comment by innovationhub-asia** [comment]
Works! Thank you very much.

**Comment by xkizilx** [comment]
Apologies for commenting on closed issue, but unfortunately dependency resolution does not work again starting from 2.7.4.10.
```
+--- it.tdlight:tdlight-java-bom:2.7.10.4
|    +--- it.tdlight:tdlight-java:2.7.10.4 (c)
|    +--- it.tdlight:tdlight-natives-linux-amd64:4.0.198 FAILED
|    +--- it.tdlight:tdlight-api-legacy:4.0.196 FAILED
```
I've noticed that `-SNAPSHOT` suffix is added to the artifact version in the pom:
https://mvn.mchv.eu/repository/mchv/it/tdlight/tdlight-api-legacy/4.0.196/tdlight-api-legacy-4.0.196.pom.
Could this be causing the issue?

**Comment by cavallium** [comment]
> Apologies for commenting on closed issue, but unfortunately dependency resolution does not work again starting from 2.7.10.4
> 
> ```
> +--- it.tdlight:tdlight-java-bom:2.7.10.4
> |    +--- it.tdlight:tdlight-java:2.7.10.4 (c)
> |    +--- it.tdlight:tdlight-natives-linux-amd64:4.0.198 FAILED
> |    +--- it.tdlight:tdlight-api-legacy:4.0.196 FAILED
> ```
> 
> I've noticed that `-SNAPSHOT` suffix is added to the artifact version in the pom: https://mvn.mchv.eu/repository/mchv/it/tdlight/tdlight-api-legacy/4.0.196/tdlight-api-legacy-4.0.196.pom. Could this be causing the issue?

Did you try with 2.7.10.7?

**Comment by xkizilx** [comment]
> Did you try with 2.7.10.7?

Yes, I've tried the latest version first and then started to downgrade.
```
+--- it.tdlight:tdlight-java-bom:2.7.10.7
|    +--- it.tdlight:tdlight-java:2.7.10.7 (c)
|    +--- it.tdlight:tdlight-natives-linux-amd64:4.0.200 FAILED
|    +--- it.tdlight:tdlight-api-legacy:4.0.198 FAILED
```
The same suffix in `it.tdlight:tdlight-api-legacy:4.0.198`
https://mvn.mchv.eu/repository/mchv/it/tdlight/tdlight-api-legacy/4.0.198/tdlight-api-legacy-4.0.198.pom
It seems that that gradle is able to find the pom in Nexus, but is failing since the version in the pom is different from one that is expected.

**Comment by cavallium** [comment]
> > Did you try with 2.7.10.7?
> 
> Yes, I've tried the latest version first and then started to downgrade.
> 
> ```
> +--- it.tdlight:tdlight-java-bom:2.7.10.7
> |    +--- it.tdlight:tdlight-java:2.7.10.7 (c)
> |    +--- it.tdlight:tdlight-natives-linux-amd64:4.0.200 FAILED
> |    +--- it.tdlight:tdlight-api-legacy:4.0.198 FAILED
> ```
> 
> The same suffix in `it.tdlight:tdlight-api-legacy:4.0.198` https://mvn.mchv.eu/repository/mchv/it/tdlight/tdlight-api-legacy/4.0.198/tdlight-api-legacy-4.0.198.pom It seems that that gradle is able to find the pom in Nexus, but is failing since the version in the pom is different from one that is expected.

Thanks, I just noticed that the flattening plugin replaces only the property ${revision}. The next version should fix it

**Comment by cavallium** [comment]
Fixed in 2.7.10.8
